src/frontend/src/icp/services/icrc.services.ts

```
@@ -343,3 +346,42 @@ export const loadDisabledIcrcTokensExchanges = async ({
 		...(icrcResult.status === 'fulfilled' && nonNullish(icrcResult.value) ? [icrcResult.value] : [])
 	]);
 };
+
+/**
+ * Checks if the owner has sufficient allowance for the spender to execute a transaction.
+ *
+ * @returns `true` if allowance is sufficient and won't expire within the buffer period, `false` otherwise
+ *
+ */
+export const hasSufficientIcrcAllowance = async ({
+	identity,
+	ledgerCanisterId,
+	owner,
+	spender,
+	amount,
+	allowanceBuffer = 30_000_000_000n
+}: {
+	identity: Identity;
+	ledgerCanisterId: CanisterIdText;
+	owner: Principal;
+	spender: Principal;
+	amount: bigint;
+	allowanceBuffer?: bigint;
+}): Promise<boolean> => {
+	try {
+		const { allowance, expires_at } = await icrcAllowance({
+			identity,
+			ledgerCanisterId,
+			owner: { owner },
+			spender: { owner: spender }
+		});
+
+		const expiredBuffer = nowInBigIntNanoSeconds() + allowanceBuffer;
+		const hasSufficientAllowance = allowance >= amount;
+		const isNotExpired = nonNullish(expires_at[0]) && expires_at[0] > expiredBuffer;
+
+		return hasSufficientAllowance && isNotExpired;
+	} catch (_: unknown) {
+		return false;
```

why would we ignore the errors? we should raise them, you never know why the error was created, right?